internal/pkg/cli/local_run.go

```diff
@@ -297,16 +301,52 @@ func (o *localRunOpts) Execute() error {
 	}
 	o.imageInfoList = append(o.imageInfoList, sidecarImageLocations...)
 
-	err = o.runPauseContainer(context.Background(), ports)
-	if err != nil {
-		return fmt.Errorf("run pause container: %w", err)
-	}
+	g, ctx := errgroup.WithContext(context.Background())
+	gotSigInt := &atomic.Bool{}
+	finishedErr := errors.New("containers stopped")
 
-	err = o.runContainers(context.Background(), o.imageInfoList, envVars)
-	if err != nil {
+	g.Go(func() (err error) {
+		defer func() {
+			// if we've received a sigint, we want to ignore
+			// any errors coming from this goroutine
+			if gotSigInt.Load() {
+				err = nil
+			}
+		}()
+
+		if err := o.runPauseContainer(ctx, ports); err != nil {
+			return fmt.Errorf("run pause container: %w", err)
+		}
+
+		if err := o.runContainers(ctx, o.imageInfoList, envVars); err != nil {
+			return err
+		}
+
+		return finishedErr
+	})
+
+	g.Go(func() error {
+		sigCh := make(chan os.Signal, 1)
+		signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
I also had a question do we need `syscall.SIGTERM`. I thought in order to captute `ctrl-c`, `SIGINT` is enough right?

Reply:
yeah, you're right that `SIGINT` is the signal specifically for ctrl+c. however, in the (unlikely) case that someone wants to run this from within a separate program, it's normal for a _process_ to send `SIGTERM` to tell the program that it should shut down (for instance, running `docker stop` sends `SIGTERM` to the container).

Reply:
that makes sense 👍